tests/test_rio_info.py

```diff
@@ -8,6 +8,361 @@
+    runner = CliRunner()
+    inputfile = str(data.join('float_nan.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--nodata', 'NaN'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.nodata != src.nodata
+
+
+def test_edit_crs_err(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--crs', 'LOL:WUT'])
+    assert result.exit_code == 2
+
+
+def test_edit_crs_epsg(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--crs', 'EPSG:32618'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs == {'init': 'epsg:32618'}
+
+
+def test_edit_crs_proj4(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--crs', '+init=epsg:32618'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs == {'init': 'epsg:32618'}
+
+
+def test_edit_crs_obj(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group,
+        ['edit-info', inputfile, '--crs', '{"init": "epsg:32618"}'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs.to_dict() == {'init': 'epsg:32618'}
+
+
+def test_edit_transform_err_not_json(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--transform', 'LOL'])
+    assert result.exit_code == 2
+
+
+def test_edit_transform_err_bad_array(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--transform', '[1,2]'])
+    assert result.exit_code == 2
+
+
+def test_edit_transform_affine(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    input_t = '[300.038, 0.0, 101985.0, 0.0, -300.042, 2826915.0]'
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--transform', input_t])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        for a, b in zip(src.transform, json.loads(input_t)):
+            assert round(a, 6) == round(b, 6)
+
+
+def test_edit_transform_gdal(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    gdal_geotransform = '[101985.0, 300.038, 0.0, 2826915.0, 0.0, -300.042]'
+    result = runner.invoke(main_group, [
+        'edit-info', inputfile, '--transform', gdal_geotransform])
+    assert result.exit_code != 0
+    assert 'not recognized as an Affine array' in result.output
+    assert gdal_geotransform in result.output
```

Comment:
Unclear what value this `assert` adds here; the above conditions ensure that it raised an error as expected.

Reply:
This only shows up as new due to the test shuffle but I removed the final `assert`.